pss/pss_test.go

```diff
@@ -47,30 +50,87 @@ func TestTrojanChunkRetrieval(t *testing.T) {
 	payload := []byte("RECOVERY CHUNK")
 	topic := trojan.NewTopic("RECOVERY")
 
-	var ch chunk.Chunk
+	var monitor *Monitor
 
-	pss := NewPss(localStore)
+	pss := NewPss(localStore, tags)
 
 	// call Send to store trojan chunk in localstore
-	if ch, err = pss.Send(ctx, testTargets, topic, payload); err != nil {
+	if monitor, err = pss.Send(ctx, testTargets, topic, payload); err != nil {
 		t.Fatal(err)
 	}
 
 	// verify store, that trojan chunk has been stored correctly
 	var storedChunk chunk.Chunk
-	if storedChunk, err = localStore.Get(ctx, chunk.ModeGetRequest, ch.Address()); err != nil {
+	if storedChunk, err = localStore.Get(ctx, chunk.ModeGetRequest, monitor.chunk.Address()); err != nil {
 		t.Fatal(err)
 	}
+	storedChunk = storedChunk.WithTagID(monitor.chunk.TagID())
 
-	if !reflect.DeepEqual(ch, storedChunk) {
+	if !reflect.DeepEqual(monitor.chunk, storedChunk) {
 		t.Fatalf("store chunk does not match sent chunk")
 	}
 
 	// check if pinning makes a difference
 
 }
 
-func newMockLocalStore(t *testing.T) *localstore.DB {
+func TestPssMonitor(t *testing.T) {
+	var err error
+	ctx := context.TODO()
+	tags := chunk.NewTags()
+	timeout := 10 * time.Second
+
+	localStore := newMockLocalStore(t, tags)
+
+	testTargets := [][]byte{
+		{57, 120},
+		{209, 156},
+		{156, 38},
+		{89, 19},
+		{22, 129}}
+	payload := []byte("RECOVERY CHUNK")
+	topic := trojan.NewTopic("RECOVERY")
+
+	var monitor *Monitor
+
+	pss := NewPss(localStore, tags)
+
+	// call Send to store trojan chunk in localstore
+	if monitor, err = pss.Send(ctx, testTargets, topic, payload); err != nil {
+		t.Fatal(err)
+	}
+
+loop:
+	for {
+		select {
+		case state := <-monitor.state:
+			if state == chunk.StateStored {
+				t.Log("message has been stored")
+			}
+			if state == chunk.StateSent {
+				t.Log("message has been sent")
+			}
+			if state == chunk.StateSynced {
+				t.Log("message has been synced")
+			}
+		case <-time.After(timeout):
+			t.Log("no message received")
+			close(monitor.state)
+			break loop
+		}
+	}
+
+	// we expect the chunk to be stored in localstore
+	// with a sent status
+	// and the total amount of chunk to be 1
+	var split, seen, synced, stored, sent, total int64 = 0, 0, 0, 1, 1, 1
```

Comment:
strange, is this passing? who is incrementing the synced to 1?
And since you increment Stored in the send function, how can it be 0?

Reply:
Removed as it does not make sense, this was passing due to the bug in checkTags